pulpcore/tasking/tasks.py

```diff
@@ -146,8 +150,10 @@ def dispatch(
         shared_resources = _validate_and_get_resources(shared_resources)
 
     # A task that is exclusive on a domain will block all tasks within that domain
+    domain_prn = get_prn(get_domain())
     domain_url = get_url(get_domain())
-    if domain_url not in exclusive_resources:
+    if not (domain_prn in exclusive_resources or domain_url in exclusive_resources):
```

Comment:
Would this make it more clear that we are looking for our domain to be on the exclusive resources list?
```suggestion
    if (domain_prn or domain_url) not in exclusive_resources:
```

Reply:
This will not work for the latter one, see:
```
>>> a = "bla"
>>> b = "blabla"
>>> 
>>> (a or b) in ["blabla"]
False
>>> (a or b)
'bla'
>>> (a or b) in ["bla"]
True
```